docs/source/guides/deploy_solver_new.rst

```diff
@@ -105,18 +128,75 @@ Schema property
+#. **fixSolution**: boolean. When ``true``, the solution method will fix the existing information in the solution when exploring the solution space.
+#. **gapAbs**: float. The maximum absolute gap allowed when considering a solution optimal.
+#. **gapRel**: float. The maximum relative gap allowed when considering a solution optimal.
+#. **threads**: integer. The number of cores that should be used in the solution method.
+
+Cornflow-client has some utility functions to help while making this configuration schema. For small solution methods, it doesn't make much sense to be creating a separated ``config.json`` schema file. For these cases, it's possible to use :py:func:`cornflow_client.schema.tools.get_empty_schema`.
+
+Instance and solution schemas
+*******************************
+
+Instance and Solution classes also require their own schemas. See their own sections for more details on how to provide them: :ref:`Instance class` and :ref:`Solution class` respectively.
+
+
+README
+--------------
+
+It is a good idea to have a description of the problem to be solved in text form. Even better if the text is accompanied by a mathematical formulation in LaTeX. This way, others can check the problem description and better understand the input data, output data and the solution methods.
+
 
 Code structure
 ------------------------------------
 
-Each app is contained ideally inside a directory. Usually, the ``Instance``, ``Solution`` and ``Solution method`` are located each one inside a separate file. The ``__init__.py`` needs to contain the Application class. Other files inside the folder include the schemas (better stored as json files).
+The following is the official structure to organize an application::
 
-In this example we put everything inside the ``__init__.py`` except the json-schema files because it was a small example.
+    my-project/
+        __init__.py
+        README.md
+        core/
+            __init__.py
+            instance.py
+            solution.py
+            experiment.py
+        schemas/
+            __init__.py
+            instance.json
+            solution.json
+            config.json
+        data/
+            data_file_1.json
+            data_file_2.json
+        solvers/
+            __init__.py
+            solver_1.py
+            solver_2.py
 
+Each app is contained ideally inside a directory.
+
+The :ref:`Instance class` is implemented in the ``my-project/core/instance.py`` file, the :ref:`Solution class` inside the ``my-project/core/solution.py``.
+
+Each :ref:`Solution method` is defined in the ``my-project/solvers`` directory: ``solver_1.py``, ``solver_2.py``, etc.
+
+The :ref:`Application class` is defined inside ``my-project/__init__.py``.
+
+Schemas are stored in the ``my-project/schemas`` folder. Finally, :ref:`Test cases` are stored in the ``my-project/data`` directory.
```

Comment:
I would change my-project to my_project because while importing, for example, in the test we can not do: `from DAG.my-project import Application`.

As it will throw an error because python modules can not use the `-` but it can use the underscore so `from DAG.my_project import Application` will work